### PR TITLE
Updated travis config file to use conda

### PR DESCRIPTION
**Pull request by tpaviot**
This branch:

* adds a conda build script

* uses conda to build smesh on travis

* adds support for both osx and linux on travis

### CODE REVIEW

File: ci/conda/build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ -DSMESH_TESTING=ON \
+ .
+
+# Build step
+make -j 4
+
+# test
+make test
+
+# Install step
```
Comment (jf---) [inline ci/conda/build.sh]: @tpaviot , could this be the issue of the failed osx built?
i guess running `make install` is required to link the libraries
let me check

**Comment by tpaviot** [comment]
No, there's only a failure on linux, becasue of the "stl binary" failing test. osx built is fine see https://travis-ci.org/tpaviot/smesh/builds/119791711

**Comment by tpaviot** [comment]
@jf--- yes maybe. I did not find any other solution/place to run ```make test```

**Comment by jf---** [comment]
i tried a few things:
- binary_relocation off -> on
- set MACOSX_RPATH to $PREFIX/lib
- run `make install` before `make test`

conda creates another env for tests, while gtests are ran in the build env
i'm pretty confident this is the source of error

but no idea how the test could be invoked separately

**Pull request merged**